File: c++/0342.cpp
// Link: https://leetcode.com/problems/power-of-four/
// Runtime: 0 ms 
// Memory: 8 MB

class Solution {
public:
    bool isPowerOfFour(int n) 
    {
        return (n > 0) && 
            (std::fabs(std::log10(n) / std::log10(4) - static_cast<int>(std::log10(n) / std::log10(4))) < std::numeric_limits<double>::epsilon());
    }
};

File: python/0342.py
# Link: https://leetcode.com/problems/power-of-four/description/
# Runtime: 0 ms
# Memory: 17.82 MB


class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        return (n > 0) and fabs(
            log10(n) / log10(4) - int(log10(n) / log10(4))
        ) < float_info.epsilon
